client/src/main/java/io/pravega/client/stream/impl/ControllerResolverFactory.java

```diff
@@ -107,24 +128,12 @@ public String getDefaultScheme() {
          * @param enableDiscovery   Whether to use the controller's discovery API.
          */
         ControllerNameResolver(final String authority, final List<InetSocketAddress> bootstrapServers,
-                                      final boolean enableDiscovery) {
+                               final boolean enableDiscovery) {
             this.authority = authority;
-            this.bootstrapServers = bootstrapServers;
+            this.bootstrapServers = new ArrayList<>(bootstrapServers);
```

Comment:
Sorry, I mistook this for java's unmodifiablelist, I guess the guava version is threadsafe. Will verify and use it.